includes/PostRestAPI.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Class SubscribeWithGoogle\WordPress\PostRestAPI
+ *
+ * @package   SubscribeWithGoogle\WordPress
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ */
+
+namespace SubscribeWithGoogle\WordPress;
+
+use WP_REST_Request;
+use WP_Error;
+
+/**
+ * Override the REST API route that leads directly to the WP Post Content
+ */
+final class PostRestAPI {
+
+
+	/**
+	 * Identifier of GoogleSignIn class.
+	 * Tests can override this.
+	 *
+	 * @var string
+	 */
+	public static $google_sign_in_class =
+	'SubscribeWithGoogle\WordPress\GoogleSignIn';
+
+	/** Adds action handlers. */
+	public function __construct() {
+		add_action( 'rest_api_init', array( __CLASS__, 'register_rest_routes' ) );
+	}
+
+	/** Registers custom REST routes. */
+	public static function register_rest_routes() {
+		register_rest_route(
+			'wp/v2',
```

Comment:
One potential issue - this overrides the standard WP REST API for fetching posts - I figured this was a good idea since the content is only valuable if it's not easily accessible, _and_ we needed a way to fetch the content asynchronously.